llsd/serde_xml.py

```diff
@@ -267,19 +277,33 @@ def parse_xml_nohdr(baseparser):
 
 
 _g_xml_formatter = None
+def _get_xml_formatter():
+    global _g_xml_formatter
+    if _g_xml_formatter is None:
+        _g_xml_formatter = LLSDXMLFormatter()
+    return _g_xml_formatter
```

Comment:
Is this class construction expensive enough to merit lazy creation?

Reply:
Perhaps not. `LLSDBaseFormatter` does construct `type_map`, requiring 19 attribute lookups, but that's not a lot of initialization. I simply extracted earlier logic from `format_xml()` to make it available to `write_xml()` as well. If you'd rather the functionality vanish entirely, I won't be distraught.

Reply:
In this case I'd prefer to eat the one-time import-time cost and simplify.

Reply:
More importantly:  it's profoundly unsafe now that self.stream is carrying state.  Threads, coroutines, etc., all break with shared instances.

Reply:
Thanks, monty, should've caught that myself.